src/lepton/jpgcoder.cc

```diff
@@ -2068,8 +2068,11 @@ void process_file(IOUtil::FileReader* reader,
           if ( errorlevel.load() < err_tresh ) {
                 if (action == comp ) {
                     fprintf(stdout, 
-                        (filetype == JPEG?  "STATS %.6lf %s %d %d %.6lf %.6lf %.6lf\n"
-                                         :  "STATS %.6lf %s %d %d %.6lf\n"),
+                        (filetype == JPEG?  "%s %.6lf %s %d %d %.6lf %.6lf %.6lf\n"
+                                         :  "%s %.6lf %s %d %d %.6lf\n"),
+                        jpegtype == JPEG_TYPE::BASELINE?      "baseline"
```

Comment:
Now it looks more heavy with sequence of `:?`. Let's replace `jpegtype -> string` with a function that has switch/case&

Reply:
No, it isn't good too. Combination of nested `?:` is bad idea in any form.

Reply:
```C
const char* jpegtype_str;
switch (jpegtype) {
  case JPEG_TYPE::BASELINE:        jpegtype_str = "baseline"; break;
  case JPEG_TYPE::PROGRESSIVE:     jpegtype_str = "progressive"; break;
  default:                         jpegtype_str = "UNKNOWN";
}
```

ok?

Reply:
Yes, it is better to wrap it into a function.